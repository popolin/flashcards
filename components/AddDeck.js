import React, { Component } from 'react';
import { connect } from 'react-redux'
import { TouchableOpacity, Text, TextInput, View, TouchableWithoutFeedback, Keyboard, StyleSheet } from 'react-native';
import { Button } from 'react-native-elements';


import * as API from '../utils/api'
import { addDeck } from "../actions/index";

class AddDeck extends Component {

    state = {
        title: ''
    }

    componentWillMount() {
        this.props.navigation.addListener('willFocus', this.willFocus);
    }

    willFocus = a => {
        console.log('TABS EVENT...', a.type, a);
        this.setState({title: ''})
    };

    createDeck = () => {
        Keyboard.dismiss()
        const { dispatch, navigation } = this.props
        const uuidv1 = require('uuid/v1');
        const key = uuidv1();
        const deck = {title: this.state.title, questions: []}

        API.addDeck(key, deck).then(() => {
            dispatch(addDeck(key, deck))
            this.setState({title: ''})
            navigation.navigate('DeckDetail', {keyDeck: key})
        })

    }

    render() {
        const { navigation, decks } = this.props
        const buttonDisabled = this.state.title.trim().length === 0
        return (
            <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
                <View style={styles.container}>
                    <Text style={styles.title}>What is the title of your new deck?</Text>
                    <TextInput
                        value={this.state.title}
                        onChangeText={(text) => this.setState({title: text})}
                        placeholder="Title"
                        onSubmitEditing={Keyboard.dismiss}
                        style={styles.input}
                    />


                    <Button onPress={this.createDeck}
                            disabled={buttonDisabled}
                            buttonStyle={{backgroundColor: 'rgba(78, 116, 289, 1)', width: null, height: 40, marginTop: 20}}
                            textStyle={{color: 'white', marginHorizontal: 20}}
                            title='Create Deck'
                    />

                </View>
            </TouchableWithoutFeedback>
        )
    }

}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#ffffff'
    },
    btn: {
        marginTop: 20,
    },
    input: {
        fontSize: 18,
        borderBottomWidth: 1,
        borderColor: 'rgba(171, 189, 219, 1)',
        marginLeft: 20,
        marginRight: 20
    },
    title: {
        padding: 20,
        fontSize: 20,
        textAlign: 'center'
    },
})

function mapStateToProps(state) {
    return {
        decks: state
    }
}

export default connect(mapStateToProps)(AddDeck)