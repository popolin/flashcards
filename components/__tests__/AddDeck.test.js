import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import AddDeck from '../AddDeck';

const middlewares = [thunk]
const mockStore = configureStore(middlewares)
const store = mockStore({})
const props = {
    navigation: {
        addListener: jest.fn()
    }
}

describe('Add Deck', () => {

    it('renders without crashing', async () => {
        store.dispatch = jest.fn();
        const rendered = renderer.create(<AddDeck {...props} store={store} />).toJSON();
        expect(rendered).toMatchSnapshot();
    });

})
