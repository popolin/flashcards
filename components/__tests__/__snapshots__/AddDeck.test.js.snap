// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Deck renders without crashing 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
  testID={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "padding": 20,
        "textAlign": "center",
      }
    }
  >
    What is the title of your new deck?
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Title"
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "rgba(171, 189, 219, 1)",
        "fontSize": 18,
        "marginLeft": 20,
        "marginRight": 20,
      }
    }
    value=""
  />
  <View
    style={
      Array [
        Object {
          "marginLeft": 15,
          "marginRight": 15,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        pointerEvents="box-only"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "padding": 12,
            },
            undefined,
            undefined,
            undefined,
            Object {
              "backgroundColor": "rgba(78, 116, 289, 1)",
              "height": 40,
              "marginTop": 20,
              "width": null,
            },
            Object {
              "backgroundColor": "#dadee0",
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                Object {
                  "color": "white",
                  "marginHorizontal": 20,
                },
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Create Deck
        </Text>
      </View>
    </View>
  </View>
</View>
`;
